From ee4a6253eb1273b3130da27661828b584574e801 Mon Sep 17 00:00:00 2001
From: Kedareswara rao Appana <appana.durga.rao@xilinx.com>
Date: Thu, 24 Sep 2015 12:47:25 +0530
Subject: [PATCH 09/77] net: ethernet: xilinx: move phy related definitions to
 common header file

This patch created a include file for Xilinx PCS PMA PHYs and moves relevant details to it.

Signed-off-by: Kedareswara ran Appana<appanad@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
[zou:Original patch taken from
https://github.com/Xilinx/linux-xlnx.git Petalinux 2015.4]
Signed-off-by: zou cao <cao.zou@windriver.com>
---
 drivers/net/ethernet/xilinx/xilinx_axienet.h      |    8 --------
 drivers/net/ethernet/xilinx/xilinx_axienet_main.c |    1 +
 drivers/net/phy/xilinx_phy.c                      |    6 ++----
 include/linux/xilinx_phy.h                        |   18 ++++++++++++++++++
 4 files changed, 21 insertions(+), 12 deletions(-)
 create mode 100644 include/linux/xilinx_phy.h

diff --git a/drivers/net/ethernet/xilinx/xilinx_axienet.h b/drivers/net/ethernet/xilinx/xilinx_axienet.h
index 63ac9a9..f99dc13 100644
--- a/drivers/net/ethernet/xilinx/xilinx_axienet.h
+++ b/drivers/net/ethernet/xilinx/xilinx_axienet.h
@@ -328,14 +328,6 @@
 
 #define XAE_MDIO_DIV_DFT		29 /* Default MDIO clock divisor */
 
-/* Defines for different options for C_PHY_TYPE parameter in Axi Ethernet IP */
-#define XAE_PHY_TYPE_MII		0
-#define XAE_PHY_TYPE_GMII		1
-#define XAE_PHY_TYPE_RGMII_1_3		2
-#define XAE_PHY_TYPE_RGMII_2_0		3
-#define XAE_PHY_TYPE_SGMII		4
-#define XAE_PHY_TYPE_1000BASE_X		5
-
  /* Total number of entries in the hardware multicast table. */
 #define XAE_MULTICAST_CAM_TABLE_NUM	4
 
diff --git a/drivers/net/ethernet/xilinx/xilinx_axienet_main.c b/drivers/net/ethernet/xilinx/xilinx_axienet_main.c
index 5d4ecb2..73ded0d 100644
--- a/drivers/net/ethernet/xilinx/xilinx_axienet_main.c
+++ b/drivers/net/ethernet/xilinx/xilinx_axienet_main.c
@@ -38,6 +38,7 @@
 #include <linux/ptp_classify.h>
 #include <linux/net_tstamp.h>
 #include <net/sock.h>
+#include <linux/xilinx_phy.h>
 
 #include "xilinx_axienet.h"
 
diff --git a/drivers/net/phy/xilinx_phy.c b/drivers/net/phy/xilinx_phy.c
index 0339dd6..2f864cc 100644
--- a/drivers/net/phy/xilinx_phy.c
+++ b/drivers/net/phy/xilinx_phy.c
@@ -21,15 +21,13 @@
 #include <linux/mii.h>
 #include <linux/phy.h>
 #include <linux/of.h>
+#include <linux/xilinx_phy.h>
 
-#define XILINX_PHY_ID			0x01740c00
-#define XILINX_PHY_ID_MASK		0xfffffff0
 #define MII_PHY_STATUS_SPD_MASK		0x0C00
 #define MII_PHY_STATUS_FULLDUPLEX	0x1000
 #define MII_PHY_STATUS_1000		0x0800
 #define MII_PHY_STATUS_100		0x0400
 #define XPCSPMA_PHY_CTRL_ISOLATE_DISABLE 0xFBFF
-#define	XPCSPMA_TYPEIS_1000BASE_X	5
 
 static int xilinxphy_read_status(struct phy_device *phydev)
 {
@@ -87,7 +85,7 @@ static int xilinxphy_read_status(struct phy_device *phydev)
 	/* For 1000BASE-X Phy Mode the speed/duplex will always be
 	 * 1000Mbps/fullduplex
 	 */
-	if (phydev->dev_flags == XPCSPMA_TYPEIS_1000BASE_X) {
+	if (phydev->dev_flags == XAE_PHY_TYPE_1000BASE_X) {
 		phydev->duplex = DUPLEX_FULL;
 		phydev->speed = SPEED_1000;
 	}
diff --git a/include/linux/xilinx_phy.h b/include/linux/xilinx_phy.h
new file mode 100644
index 0000000..56cca81
--- /dev/null
+++ b/include/linux/xilinx_phy.h
@@ -0,0 +1,18 @@
+#ifndef _XILINX_PHY_H
+#define _XILINX_PHY_H
+
+/* Mask used for ID comparisons */
+#define XILINX_PHY_ID_MASK		0xfffffff0
+
+/* Known PHY IDs */
+#define XILINX_PHY_ID			0x01740c00
+
+/* struct phy_device dev_flags definitions */
+#define XAE_PHY_TYPE_MII		0
+#define XAE_PHY_TYPE_GMII		1
+#define XAE_PHY_TYPE_RGMII_1_3		2
+#define XAE_PHY_TYPE_RGMII_2_0		3
+#define XAE_PHY_TYPE_SGMII		4
+#define XAE_PHY_TYPE_1000BASE_X		5
+
+#endif /* _XILINX_PHY_H */
-- 
1.7.5.4

